aten/src/ATen/native/cuda/group_norm_kernel.cu

```diff
@@ -360,6 +360,39 @@ __global__ void ComputeInternalGradientsCUDAKernel(
   }
 }
 
+template <typename T>
+__global__ void ComputeInternalGradientsCUDAKernelNHWC(
+    int64_t HxW,
+    int64_t C,
+    const T* dY,
+    const T* X,
+    acc_type<T, true>* ds,
+    acc_type<T, true>* db) {
+  using T_ACC = acc_type<T, true>;
+  const int64_t n = blockIdx.x / C;
+  const int64_t c = blockIdx.x % C;
+  T_ACC sum1 = 0;
+  T_ACC sum2 = 0;
+  for (int64_t hw = threadIdx.x; hw < HxW; hw += blockDim.x) {
+    const int64_t index = n * HxW * C + hw * C + c;
+    sum1 += static_cast<T_ACC>(dY[index]) * static_cast<T_ACC>(X[index]);
+    sum2 += static_cast<T_ACC>(dY[index]);
+  }
+  if (blockDim.x <= C10_WARP_SIZE) {
+    sum1 = cuda_utils::WarpReduceSum<T_ACC>(sum1);
+    sum2 = cuda_utils::WarpReduceSum<T_ACC>(sum2);
+  } else {
+    __shared__ T_ACC ds_shared[C10_WARP_SIZE];
+    __shared__ T_ACC db_shared[C10_WARP_SIZE];
+    sum1 = cuda_utils::BlockReduceSum<T_ACC>(sum1, ds_shared);
+    sum2 = cuda_utils::BlockReduceSum<T_ACC>(sum2, db_shared);
+  }
+  if (threadIdx.x == 0) {
+    ds[blockIdx.x] = sum1;
+    db[blockIdx.x] = sum2;
+  }
+}
+
```

Comment:
Do we need a new function for this if the only difference is the indexing math?

Reply:
I was thinking about that. How should we approach this, would a function arg specifying the memory format be good enough?

Reply:
True. We could probably also remove the NHWC kernel in the fwd pass